src/libmongoc/doc/mongoc_client_get_handshake_description.rst

```diff
@@ -0,0 +1,50 @@
+:man_page: mongoc_client_get_handshake_description
+
+mongoc_client_get_handshake_description()
+=========================================
+
+Synopsis
+--------
+
+.. code-block:: c
+
+   mongoc_server_description_t *
+   mongoc_client_get_handshake_description (mongoc_client_t *client,
+                                            uint32_t server_id,
+                                            bson_t *opts,
+                                            bson_error_t *error);
+
+Returns a description constructed from the initial handshake response to a server.
+
+Description
+-----------
+
+:symbol:`mongoc_client_get_handshake_description` is distinct from :symbol:`mongoc_client_get_server_description`. :symbol:`mongoc_client_get_server_description` returns a server description constructed from monitoring, which may differ from the server description constructed from the connection handshake.
+
+:symbol:`mongoc_client_get_handshake_description` does not attempt to establish a connection to the server if a connection was not already established. It will complete authentication on a single-threaded client if the connection has not yet been used by an operation.
```

Comment:
Ah, does this pertain to monitoring not using auth at all? But on single-threaded, since it's also the application socket auth ends up being completed?

I'm unsure if this current behavior would be problematic, but I think anything we can do to make it consistent between modes of operation would be welcome. It simplifies the documentation and can avoid a subtle impact (which we might not realize now) down the line.

Reply:
> Ah, does this pertain to monitoring not using auth at all?

Yes. Monitoring, even for single-threaded, will not authenticate connections. In single-threaded, a connection completes authentication the first time it is used for an operation when `mongoc_cluster_fetch_stream_single` is called.

> I'm unsure if this current behavior would be problematic, but I think anything we can do to make it consistent between modes of operation would be welcome. It simplifies the documentation and can avoid a subtle impact (which we might not realize now) down the line.

I agree it seems simpler to reason about `mongoc_client_get_handshake_description` if it also attempts to establish a connection.